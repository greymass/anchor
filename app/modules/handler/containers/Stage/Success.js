// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import { Grid, Header, Icon, Message, Table } from 'semantic-ui-react';

import PromptStageCallback from './Callback';
import ExplorerLink from '../../../../shared/containers/Global/Blockchain/ExplorerLink';

class PromptStageSuccess extends Component<Props> {
  render() {
    const {
      blockchain,
      callbacking,
      hasBroadcast,
      hasForegroundCallback,
      prompt,
      settings,
      t,
    } = this.props;
    const {
      response,
      signed,
    } = prompt;
    const tx = response || signed;
    // Remove any UUID from callback URL
    const callbackURL = (prompt && prompt.callbackURL)
      ? prompt.callbackURL.replace(/[0-9a-f]{8}-?[0-9a-f]{4}-?[1-5][0-9a-f]{3}-?[89ab][0-9a-f]{3}-?[0-9a-f]{12}/ig, '')
      : '';
    // Retrieve block number and transaction ID
    let block_num;
    let transaction_id;
    if (prompt.callbackPayload) {
      block_num = prompt.callbackPayload.bn;
      transaction_id = prompt.callbackPayload.tx;
    } else if (tx && tx.processed) {
      block_num = tx.processed.block_num;
      transaction_id = tx.transaction_id;
    }
    return (
      <Grid>
        {(hasForegroundCallback)
          ? (
            <Grid.Column width={hasBroadcast ? 7 : 16}>
              <PromptStageCallback
                blockchain={blockchain}
                callbacking={callbacking}
                prompt={prompt}
                settings={settings}
                singleColumn
              />
            </Grid.Column>
          )
          : false
        }
        {(hasBroadcast || !hasForegroundCallback)
          ? (
            <Grid.Column width={hasForegroundCallback ? 9 : 16}>
              <Header
                size="huge"
              >
                <Icon color="green" name="check circle outline" />
                <Header.Content>
                  {t('handler_containers_stage_success_header')}
                  <Header.Subheader>
                    {t(
                      'handler_containers_stage_success_subheader',
                      {
                        responseProcessed: response && response.processed ?
                          t('handler_containers_stage_success_subheader_blockchain') :
                          t('handler_containers_stage_success_subheader_callback_service'),
                      }
                    )}
                  </Header.Subheader>
                </Header.Content>
              </Header>
              <Table
                definition
                style={{
                  display: 'block',
                  overflowX: 'scroll',
                }}
              >
                <Table.Body>
                  {(transaction_id)
                    ? (
                      <Table.Row>
                        <Table.Cell collapsing>Transaction ID</Table.Cell>
                        <Table.Cell>
                          <ExplorerLink
                            content={transaction_id}
                            linkData={transaction_id}
                            linkBlockId={(block_num) ? block_num : false}
                            linkType="txid"
                          />
                        </Table.Cell>
                      </Table.Row>
                    )
                    : false
                  }
                  <Table.Row>
                    <Table.Cell collapsing>{t('handler_containers_stage_success_table_cell_two')}</Table.Cell>
                    <Table.Cell>
                      {(response && response.processed)
                        ? prompt.endpoint
                        : callbackURL
                      }
                    </Table.Cell>
                  </Table.Row>
                </Table.Body>
              </Table>
              <Message
                color="grey"
                content={(
                  <React.Fragment>
                    <p>
                      {t('handler_containers_stage_success_message_one')}
                    </p>
                  </React.Fragment>
                )}
                header={t('handler_containers_stage_success_message_one_header')}
                icon="info circle"
                size="large"
              />
            </Grid.Column>
          )
          : false
        }
      </Grid>
    );
  }
}

function mapStateToProps(state) {
  return {
    prompt: state.prompt,
  };
}

export default compose(
  withTranslation('handler'),
  connect(mapStateToProps)
)(PromptStageSuccess);
