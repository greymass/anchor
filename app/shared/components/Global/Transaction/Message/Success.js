// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button, Header, Icon, Message, Modal, Segment } from 'semantic-ui-react';

import DangerLink from '../../Modal/DangerLink';

class GlobalTransactionMessageSuccess extends Component<Props> {
  render() {
    const {
      onClose,
      t,
      transaction,
      transactions
    } = this.props;
    const links = [];
    if (transaction) {
      links.push(<DangerLink
        content={transaction.transaction_id}
        link={`https://eospark.com/MainNet/tx/${transaction.transaction_id}`}
      />);
    }
    if (transactions) {
      transactions.map((tx) => {
        links.push(<DangerLink
          content={tx.transaction_id}
          link={`https://eospark.com/MainNet/tx/${tx.transaction_id}`}
        />);
      })
    }
    return (
      <Segment basic>
        <Header
          content={t('global_transaction_complete_title')}
          icon="checkmark"
          size="large"
        />
        <Modal.Content>
          <p>{t('global_transaction_complete_message')}</p>
          <Segment padded textAlign="center">
            <p>txids</p>
            {links.map((link, idx) => (
              <p key={idx}>
                #{idx+1}
                {' - '}
                {link}
              </p>
            ))}
            <p>(linked to eospark.com)</p>
          </Segment>
          <Message
            icon
            size="large"
            warning
          >
            <Icon
              name="info circle"
            />
            <Message.Content>
              <Message.Header>{t('global_transaction_complete_warning_title')}</Message.Header>
              {t('global_transaction_complete_warning_message')}
            </Message.Content>
          </Message>
        </Modal.Content>
        <Modal.Actions>
          <Segment basic clearing>
            <Button color="green" floated="right" onClick={onClose}>
              <Icon name="checkmark" /> {t('close')}
            </Button>
          </Segment>
        </Modal.Actions>
      </Segment>

    );
  }
}

export default translate('global')(GlobalTransactionMessageSuccess);
