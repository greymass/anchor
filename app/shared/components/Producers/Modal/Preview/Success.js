// @flow
import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import { Button, Header, Icon, Modal, Segment } from 'semantic-ui-react';
import DangerLink from '../../../Global/Modal/DangerLink';

export default class ProducersVotingPreviewSuccess extends Component<Props> {
  render() {
    const {
      lastTransaction,
      onClose
    } = this.props;
    return (
      <I18n ns="producers">
        {
          (t) => (
            <Segment>
              <Header icon="checkmark" content={t('producer_voter_preview_transaction_successful_title')} />
              <Modal.Content>
                <h3>
                  {t('producer_voter_preview_transaction_successful_message')}
                </h3>
                <Segment basic padded textAlign="center">
                  <DangerLink
                    content={lastTransaction.transaction_id}
                    link={`https://eospark.com/MainNet/tx/${lastTransaction.transaction_id}`}
                  />
                </Segment>
              </Modal.Content>
              <Modal.Actions>
                <Button color="green" onClick={onClose}>
                  <Icon name="checkmark" /> {t('producer_voter_preview_close')}
                </Button>
              </Modal.Actions>
            </Segment>
          )
        }
      </I18n>
    );
  }
}
