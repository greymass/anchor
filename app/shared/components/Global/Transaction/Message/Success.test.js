import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionMessageSuccess } from './Success';

const mockProps = {
  blockExplorers: {},
  hideClose: false,
  onClose: () => null,
  settings: {},
  t: text => text,
  transaction: {
    processed: {
      block_num: 8,
    },
    transaction_id: 'transaction_id'
  },
  transactions: [],
};

describe('GlobalTransactionMessageSuccess', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionMessageSuccess {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

