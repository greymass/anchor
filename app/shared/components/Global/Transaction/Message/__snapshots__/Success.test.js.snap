// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionMessageSuccess renders correctly 1`] = `
<Segment
  basic={true}
>
  <Header
    content="global_transaction_complete_title"
    icon="checkmark"
    size="large"
  />
  <ModalContent>
    <p>
      global_transaction_complete_message
    </p>
    <Segment
      padded={true}
      textAlign="center"
    >
      <p>
        txids
      </p>
      <p
        key="0"
      >
        #
        1
         - 
        <GlobalModalExplorerLink
          blockExplorers={Object {}}
          content="transact...ction_id"
          linkBlockId={8}
          linkData="transaction_id"
          linkType="txid"
          settings={Object {}}
        />
      </p>
      <p>
        global_transaction_complete_link_to undefined
      </p>
    </Segment>
    <Message
      icon={true}
      size="large"
      warning={true}
    >
      <Icon
        as="i"
        name="info circle"
      />
      <MessageContent>
        <MessageHeader>
          global_transaction_complete_warning_title
        </MessageHeader>
        global_transaction_complete_warning_message
      </MessageContent>
    </Message>
  </ModalContent>
  <ModalActions>
    <Segment
      basic={true}
      clearing={true}
    >
      <Button
        as="button"
        color="green"
        floated="right"
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="checkmark"
        />
         
        close
      </Button>
    </Segment>
  </ModalActions>
</Segment>
`;
